oshi-core/src/main/java/oshi/util/ParseUtil.java

```diff
@@ -660,7 +663,7 @@ public static long[] parseStringToLongArray(String s, int[] indices, int length,
         boolean numeric = true;
         boolean numberFound = false; // ignore nonnumeric at end
         boolean dashSeen = false; // to flag uuids as nonnumeric
-        while (--charIndex > 0 && parsedIndex >= 0) {
+        while (charIndex-- > 0 && parsedIndex >= 0) {
```

Comment:
The change in ordering here doesn't make a difference for this conditional, but breaks the `charAt()` call below.

Reply:
update: 
1. use `--charIndex >= 0` instead of `--charIndex > 0` in the while loop condition
2. add a test case